### PR TITLE
We can now configure the worker runtime limits

### PR DESCRIPTION
**Pull request by annando**
When using the cooldown functionality, the overall runtime of a process can reach the defined limits. Then it does make sense to adjust these previously hardcoded values.

### CODE REVIEW

File: static/defaults.config.php [hunk 1]
```diff
@@ -650,6 +650,16 @@
...
+		// worker_max_duration (Array)
+		// Maximum runtime per priority. Worker processes that exceed this runtime will be terminated.
+		'worker_max_duration' => [
+			PRIORITY_CRITICAL   => 720,
+			PRIORITY_HIGH       => 10,
+			PRIORITY_MEDIUM     => 60,
+			PRIORITY_LOW        => 180,
+			PRIORITY_NEGLIGIBLE => 720
+		],
+
```
Comment (Quix0r) [inline static/defaults.config.php:662]: Maybe an idea for later PR: Move `PRIORITY_*` constants to `Worker` class?

**Review by Quix0r — APPROVED** [review]
No logic errors seen.

File: static/defaults.config.php (hunk 1 above)
Comment (MrPetovan) [reply static/defaults.config.php:662]: I second that, like all constants should be moved out of of `boot.php`. I'd prefer if it was done before these naked constants are used in a node's configuration file beyond the defaults.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: static/defaults.config.php (hunk 1 above)
Comment (annando) [reply static/defaults.config.php:662]: It is never a problem to deprecate old constants, we did this before.

Comment (MrPetovan) [reply static/defaults.config.php:662]: When they only are used in code, it's fine, but when they're supposed to be used in user-edited config files, then we don't have complete control and we should tread even lighter.

**Comment by Quix0r** [comment]
@MrPetovan I already did it for the `GRAVITY_*` constants and moved them to `Friendica\Model\Item` class.

File: static/defaults.config.php (hunk 1 above)
Comment (Quix0r) [reply static/defaults.config.php:662]: So to have both worlds united:
- constants used in configuration file, only deprecate them and name the new place
- constants used in other code, move them (including addon's code) to new place.

Can we agree on this?

**Comment by MrPetovan** [comment]
Yes and that was good, but these constants weren't used in user-edited configuration files. If they did, it would have broken the node of the admins who would have used these constants.

File: static/defaults.config.php (hunk 1 above)
Comment (MrPetovan) [reply static/defaults.config.php:662]: Apparently that's what @annando went with, keeping the global constants for the configuration file and creating new class constants?

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**